web/react/components/post_body.jsx

```diff
@@ -53,6 +57,46 @@ export default class PostBody extends React.Component {
         this.setState({links: linkData.links, message: linkData.text});
     }
 
+    createEmbed(link) {
+        let embed = this.createYoutubeEmbed(link);
+
+        if (embed != null) {
+            return embed;
+        }
+
+        embed = this.createGifEmbed(link);
+
+        return embed;
+    }
+
+    loadGif(src) {
+        const gif = new Image();
```

Comment:
Won't this send multiple requests for the gif to load since render is sometimes called multiple times in a short period?